### ISSUE TITLE
Clang-format: Incorrect removal `;` by RemoveSemicolon

### ISSUE THREAD

**Issue by JVApen** [llvm/llvm-project#61188]
```cpp
   class [[deprecated("")]] C
   {
      enum E
      {
      };
   };
```
The `;` of the class is removed by Clang-Format using `RemoveSemicolon: true`.

Encountered on Clang-Format 16-rc2

**Comment by JVApen** [comment]
Problem also occurs on:
````
   class [[deprecated("")]] C
   {
      constexpr static const auto C = 0;
   };
````
I'm not sure the content of the class is relevant, though without it, it keeps the `;`

**Comment by owenca** [comment]
https://reviews.llvm.org/D145344

**Comment by JVApen** [comment]
Thanks for being this fast, can this go to 16.0 or 16.1?

**Comment by owenca** [comment]
The due date for 16.0.0 will be in a couple of days, so the fix will probably be in 16.0.1 (if the reviews agree to backport it).

**Comment by mydeveloperday** [comment]
I'd agree to a backport. (review accpted)
